### PR TITLE
Implemented Navigation Drawer and Tabs 

### PR DESCRIPTION
**Pull request by pandilladivya**
**Implemented Navigation drawer with the following menu items**
-

- Events ( Users can view all the events happening either in a list or map view)
- Starred Events ( Users can view the bookmarked/starred events )
- My Schedule ( Users can view the events they are volunteering )
- Requests ( Users see the swap requests / notifications in this fragment )
- Navigate me ( Can navigate to a particular event / session )
- Profile ( Users can see their profile )

**Implemented Tabs for the Events Page**
- 

- ListView ( Users can see the events in a list format )
- MapView ( Events are marked on the map )

Note: I've created a basic version. Will modify UI/Menu items list later on if required. 🙂 

The following is the gif of the work done in this PR.

 
menu

### CODE REVIEW

**Comment by pandilladivya** [comment]
Kindly Review
@sandarumk @HimanshiJain 

Thank you!
🙂

**Comment by HimanshiJain** [comment]
@pandilladivya I think the among the map view and list view, the selected option must be highlighted.

**Comment by MeepyMay** [comment]
@pandilladivya Switch to 1.0 branch please. :)

**Comment by HimanshiJain** [comment]
Remove the "Requests" option from the navigation drawer as decided in discussion with May.

File: app/src/main/java/vola/systers/com/volunteers_android/Menu.java [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+
+public class Menu extends ActionBarActivity {
+    private DrawerLayout mDrawerLayout;
+    private ListView mDrawerList;
+    private ActionBarDrawerToggle mDrawerToggle;
+
+    // nav drawer title
+    private CharSequence mDrawerTitle;
+
+    // used to store app title
```
Comment (HimanshiJain) [inline app/src/main/java/vola/systers/com/volunteers_android/Menu.java]: Variables should be self-explanatory. No comments needed here. Make variables self-explanatory wherever required.

File: app/src/main/java/vola/systers/com/volunteers_android/Menu.java [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+            case 0:
+                getSupportActionBar().setIcon(R.drawable.events);
+                fragment = new EventsFragment();
+                break;
+            case 1:getSupportActionBar().setIcon(R.drawable.star);
+                fragment = new ProfileFragment();
+                break;
+            case 2:
+                getSupportActionBar().setIcon(R.drawable.schedule);
+                fragment = new ProfileFragment();
```
Comment (HimanshiJain) [inline app/src/main/java/vola/systers/com/volunteers_android/Menu.java]: I am assuming that you have added ProfileFragment for all options  just to show navigation drawer and this will change with further work.

File: app/src/main/res/layout/activity_signin.xml [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+    <TextView android:id="@+id/link_signup"
+        android:layout_width="fill_parent"
+        android:layout_height="wrap_content"
+        android:layout_marginBottom="6dp"
+        android:text="Don't have an account? Sign up here"
+        android:gravity="center"
+        android:textColor="@color/text_black"
+        android:textSize="16dip"/>
+
+    <TextView android:id="@+id/link_skip"
```
Comment (HimanshiJain) [inline app/src/main/res/layout/activity_signin.xml]: Could find no intent for this textview.

File: app/src/main/java/vola/systers/com/volunteers_android/EventsFragment.java [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+        }
+        return rootView;
+    }
+
+
+
+    private class MyPagerAdapter extends FragmentPagerAdapter {
+
+        public final int PAGE_COUNT = 2;
+
```
Comment (pandilladivya) [inline app/src/main/java/vola/systers/com/volunteers_android/EventsFragment.java]: Okay.

File: app/src/main/java/vola/systers/com/volunteers_android/Menu.java (hunk 2 above)
Comment (pandilladivya) [inline app/src/main/java/vola/systers/com/volunteers_android/Menu.java]: yes.

File: app/src/main/java/vola/systers/com/volunteers_android/Menu.java [hunk 5]
```diff
@@ -0,0 +1,229 @@
...
+                break;
+            case 4:
+                getSupportActionBar().setIcon(R.drawable.navigate);
+                fragment = new ProfileFragment();
+                break;
+            case 5:
+                getSupportActionBar().setIcon(R.drawable.profile);
+                fragment = new ProfileFragment();
+                break;
+
```
Comment (pandilladivya) [inline app/src/main/java/vola/systers/com/volunteers_android/Menu.java]: Will Remove.

**Comment by pandilladivya** [comment]
@HimanshiJain I've made all the changes.
Kindly review.
Thanks 😊

File: app/src/main/java/vola/systers/com/volunteers_android/SignInActivity.java [hunk 6]
```diff
@@ -158,4 +211,9 @@ private void hideProgressDialog() {
             mProgressDialog.hide();
         }
     }
+
+    public void onSkipClicked(View view) {
```
Comment (HimanshiJain) [inline app/src/main/java/vola/systers/com/volunteers_android/SignInActivity.java]: Do you think it is better to manage click for this using onClickListener Interface?(Since we already have a function for this interface.) We prefer using onClick implementation through XML when there are no more than one view to be handled for clicks.

File: app/src/main/res/layout/custom_tab.xml [hunk 7]
```diff
@@ -23,6 +23,6 @@
         android:maxLines="1"
         android:textAllCaps="false"
         android:textColor="@color/colorPrimaryDark"
-        android:textSize="12sp"
-        tools:text="Recents" />
+        android:textSize="@dimen/textNavItem"
```
Comment (HimanshiJain) [inline app/src/main/res/layout/custom_tab.xml]: Should it be text_nav_item? Like other dimension names?

File: app/src/main/res/layout/drawer_list_item.xml [hunk 8]
```diff
@@ -1,37 +1,37 @@
...
         android:layout_width="wrap_content"
         android:layout_height="match_parent"
-        android:layout_toRightOf="@id/icon"
+        android:layout_toRightOf="@id/nav_list_icon"
         android:minHeight="?android:attr/listPreferredItemHeightSmall"
         android:textAppearance="?android:attr/textAppearanceListItemSmall"
         android:textColor="@color/colorPrimaryDark"
         android:gravity="center_vertical"
-        android:paddingRight="40dp"/>
+        android:paddingRight="@dimen/dimenMarginRight"/>
```
Comment (HimanshiJain) [inline app/src/main/res/layout/drawer_list_item.xml]: Same as above and check for other dimension names too.

File: app/src/main/java/vola/systers/com/volunteers_android/EventsFragment.java [hunk 9]
```diff
@@ -43,34 +38,29 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
...
-
-
     private class MyPagerAdapter extends FragmentPagerAdapter {
 
         public final int PAGE_COUNT = 2;
-
         private final String[] mTabsTitle = {"ListView", "MapView"};
-
-        public MyPagerAdapter(FragmentManager fm) {
+        public MyPagerAdapter(FragmentManager fm)
```
Comment (HimanshiJain) [inline app/src/main/java/vola/systers/com/volunteers_android/EventsFragment.java]: Why this change?

File: app/src/main/res/layout/custom_tab.xml (hunk 7 above)
Comment (pandilladivya) [inline app/src/main/res/layout/custom_tab.xml]: Yeah. I'll make everything consistent.

File: app/src/main/res/layout/drawer_list_item.xml (hunk 8 above)
Comment (pandilladivya) [inline app/src/main/res/layout/drawer_list_item.xml]: Okay.

File: app/src/main/java/vola/systers/com/volunteers_android/EventsFragment.java (hunk 9 above)
Comment (pandilladivya) [inline app/src/main/java/vola/systers/com/volunteers_android/EventsFragment.java]: This was by mistake. I'll change.

File: app/src/main/java/vola/systers/com/volunteers_android/SignInActivity.java (hunk 6 above)
Comment (pandilladivya) [inline app/src/main/java/vola/systers/com/volunteers_android/SignInActivity.java]: Yeah. True

**Comment by pandilladivya** [comment]
@HimanshiJain I've made suggested changes.
Kindly review and let me know if there are anymore changes.
Thank you  ☺️ 😊

File: app/src/main/res/layout/custom_tab.xml (hunk 7 above)
Comment (HimanshiJain) [inline app/src/main/res/layout/custom_tab.xml]: Have you checked if the convention is text_nav_item or textNavItem for XML? I think it is text_nav_item.

**Comment by HimanshiJain** [comment]
And one last thing. Since you have merged the facebook integration branch with this, don't you think that after successful login it should go to the homepage? It isn't right now!
I think these changes are all I think for this PR.

File: app/src/main/res/layout/custom_tab.xml (hunk 7 above)
Comment (pandilladivya) [inline app/src/main/res/layout/custom_tab.xml]: @HimanshiJain 
I've referred to some coding practices. 
ref: https://stackoverflow.com/questions/442529/is-there-a-standard-naming-convention-for-xml-elements

So I've used Camel casing. 
Kindly let me know if i have to change it.

Comment (HimanshiJain) [inline app/src/main/res/layout/custom_tab.xml]: So I don't think there is any hard and fast rule. We can use both. But generally I think "text_nav_item" type is used. Like you yourself has named the strings similarly(for eg. skip_login). 
This also seems a nice resource: https://github.com/futurice/android-best-practices#resources . It's purely upon you whatever you think. or we can ask @sandarumk to suggest.

**Comment by pandilladivya** [comment]
@HimanshiJain 
True.
I've amended that code to navigate to the home page on successful sign to my previous commit.
:)

File: app/src/main/res/layout/custom_tab.xml (hunk 7 above)
Comment (pandilladivya) [inline app/src/main/res/layout/custom_tab.xml]: okay.

File: app/src/main/res/values/strings.xml [hunk 10]
```diff
@@ -6,8 +6,45 @@
...
+    <string name="map_view">MAP VIEW</string>
+    <string name="desc_list_item_icon">Item Icon</string>
+    <string name="recents">Recents</string>
+
+    <!-- Nav Drawer Menu Items -->
+    <string-array name="nav_drawer_items">
+        <item>Events</item>
+        <item>Starred Events</item>
+        <item>My Schedule</item>
+        <item>Requests</item>
```
Comment (HimanshiJain) [inline app/src/main/res/values/strings.xml]: Remove requests? As discussed with May.

File: app/src/main/java/vola/systers/com/volunteers_android/SignInActivity.java [hunk 11]
```diff
@@ -50,14 +65,54 @@ protected void onCreate(Bundle savedInstanceState) {
...
+    private void googleSignIn() {
+
         Intent signInIntent = Auth.GoogleSignInApi.getSignInIntent(mGoogleApiClient);
-        startActivityForResult(signInIntent, RC_SIGN_IN);
+        startActivityForResult(signInIntent, RC_GOOGLE_SIGN_IN);
+
     }
 
-    private void handleSignInResult(GoogleSignInResult result) {
+    public void onSkipClicked() {
```
Comment (HimanshiJain) [inline app/src/main/java/vola/systers/com/volunteers_android/SignInActivity.java]: How are you keeping a track if the user has logged in or not(through any method- Google/FB/Manual)? And how do you plan to manage login if the user has skipped login? Can't see any such option in the Navigation Drawer or the screen appearing each time I open the app as you have marked the Menu activity as Default in the manifest.

File: app/src/main/res/values/strings.xml (hunk 10 above)
Comment (pandilladivya) [inline app/src/main/res/values/strings.xml]: Yeah. Removed them now. :)

**Comment by sandarumk** [comment]
Also resolve the conflicts too.

File: app/src/main/java/vola/systers/com/volunteers_android/SignInActivity.java (hunk 11 above)
Comment (pandilladivya) [inline app/src/main/java/vola/systers/com/volunteers_android/SignInActivity.java]: I'll be doing that in future as I've already discussed with you 😊

**Comment by pandilladivya** [comment]
@sandarumk @HimanshiJain Made suggested Changes.
Please Let me know if I have to do anything more.
Thanks 😊

**Comment by sandarumk** [comment]
Use string file for the hard coded strings.
